libcudacxx/include/cuda/std/detail/libcxx/include/cmath

```diff
@@ -612,6 +617,146 @@ hypot(_A1 __lcpp_x, _A2 __lcpp_y, _A3 __lcpp_z) noexcept
 #define _LIBCUDACXX_CONSTEXPR_AFTER_CXX11_COMPLEX
 #endif // _LIBCUDACXX_HAS_NO_CONSTEXPR_COMPLEX_OPERATIONS
 
+#ifdef __cuda_std__
+inline _LIBCUDACXX_INLINE_VISIBILITY
+__half sin(__half __v)
+{
+    NV_IF_ELSE_TARGET(NV_IS_DEVICE,
+        (return hsin(__v);),
```

Comment:
question: Should these be fully qualified? 
```suggestion
        (return ::hsin(__v);),
```

Reply:
I don't think so; this is called on a variable of a very specific case, so no unexpected ADL should be going on. Let's have @miscco provide a second opinion.

Reply:
In general I believe we should try to fully qualify function calls whenever possible.

In this case I am potentially worried about a user defining their own `hsin` function and pulling that in.